STAR-Delegation/draft-ietf-acme-star-delegation.md

```diff
@@ -671,6 +671,90 @@ The initial contents of this registry are the extensions defined by the JSON Sch
+IdO, whose security relies on the correct handling of the associated key pair.
+When a compromise of the private key is detected, the delegate MUST use the
+account deactivation procedures defined in Section 7.3.6 of {{!RFC8555}}.
+
+The second part is covered by the act of checking an NDC's certificate request
+against the intended CSR template.  The steps of shaping the CSR template
+correctly, selecting the right CSR template to check against the presented CSR,
+and making sure that the presented CSR matches the selected CSR template are
+all security relevant.
+
+## New ACME Channels
+
+Using the model established in Section 10.1 of {{!RFC8555}}, we can decompose
+the interactions of the basic delegation workflow as shown in
+{{fig-sec-channels}}.
+
+~~~ goat
+{::include art/channels.ascii-art}
+~~~
+{: #fig-sec-channels title="Delegation Channels Topology"}
+
+The considerations regarding the security of the ACME Channel and Validation
+Channel discussed in {{!RFC8555}} apply verbatim to the IdO/ACME server leg.
+The same can be said for the ACME channel on the NDC/IdO leg.  A slightly
+different set of considerations apply to the ACME Channel between NDC and ACME
+server, which consists of a subset of the ACME interface comprising two API
+endpoints: the unauthenticated certificate retrieval and, potentially, non-STAR
+revocation via certificate private key.  No specific security consideration
+apply to the former, but the privacy considerations in Section 6.3 of
+{{!RFC8739}} do.  With regards to the latter, it should be noted that there is
+currently no means for an IdO to disable authorising revocation based on
+certificate private keys.  So, in theory, an NDC could use the revocation API
+directly with the ACME server, therefore by-passing the IdO.  The NDC SHOULD
+NOT directly use the revocation interface exposed by the ACME server unless
+failing to do so would compromise the overall security, for example if the
+certificate private key is compromised and the IdO is not currently reachable.
+
+All other security considerations from {{!RFC8555}} and {{!RFC8739}} apply
+as-is to the delegation topology.
+
+## Limiting the scope of the delegation
+{: #sec-limit-dele }
+
+Any mechanisms an IdO can use to limit the scope of the delegation that seeks
+to be universally available needs to be fully integrated with {{?RFC6125}}
+identity verification.  Unfortunately, {{?RFC6125}} does not define a standard
+way to constrain the identity verification process based on information
+contained in the presented certificate: in fact, the algorithm succeeds if the
+sets of acceptable and presented identifiers have non-empty intersection.  So,
+in an open system there is no standardised way to limit the scope of a
+delegation.
+
+In closed systems, however, a tighter control could be exerted via a locally
+defined {{?RFC6125}} profile.  For example, using a private X.509 extension
```

Comment:
I don't think it's interesting to discuss private X.509 extensions that are very unlikely to happen. There are very few - probably zero - organizations that will go to the trouble of extending X.509 for delegation of internal certificates.

Reply:
mmm, that is not my experience.  But in any case it's just a suggestion to think about how one could go ahead and achieve that (in closed or semi-closed ecosystems), if they really wanted to.

Reply:
OK.  Removing the above, this would go too.